### DISCUSSION TITLE
clarification on pdm install/sync --dev

### DISCUSSION THREAD

**Discussion by deronnax** [pdm-project/pdm#3017] [Q&A]
If I got it right, `--dev` "only" usage is as a modifier for `--group`, so `--group` actually means "dev group" and `pdm sync --dev` has no meaning, right?

**Comment by TayfurYldz** [comment]
`--dev` does have independent meaning; it is not merely a modifier for `--group`.

In the current group-selection code, `--dev` sets `dev=True`. During translation, that causes **all declared development dependency groups** to be added to the selected group set. Conversely, `--prod` sets `dev=False`, which prevents development groups from being selected.

`--group/-G` is broader: it names particular groups. Those can be optional-dependency groups, or development/dependency groups (the latter are selected with dev semantics). So these are different operations:

* `pdm sync --dev`: include the project's development dependency groups (plus the default group unless `--no-default` is used).
* `pdm sync -G docs`: explicitly select the named `docs` group in addition to the default selection.
* `pdm sync --prod`: exclude development dependencies.
* `pdm sync --prod -G <dev-group>`: rejected, because explicitly asking for a dev group conflicts with production-only selection.

There is one subtle default that can make this confusing: when neither `--dev` nor `--prod` is specified, the selection logic normally treats dev dependencies as included. Also, when the selection is otherwise completely unset and a lockfile records groups, PDM can use the lockfile's group set.

So `--dev` can appear redundant in common cases where dev groups are already included by default, but it is a real selector with defined behavior rather than syntax attached to `--group`.

**Reply by deronnax** [reply]
That's LLM answer, isn't it? If it is – which I really believe – this is a really bad behavior from you. You are making the internet a worst place. If I wanted an LLM answer, I could easily get it myself. That's silly to think you provided any value with your answer.
